app/helpers/script_helper.rb

```diff
@@ -7,7 +7,12 @@ def javascript_include_tag_without_preload(...)
   end
 
   def javascript_packs_tag_once(*names, **attributes)
-    @scripts = @scripts.to_h.merge(names.index_with(attributes))
+    scripts = RequestStore.store[:scripts]
+    if scripts
+      RequestStore.store[:scripts].merge!(names.index_with(attributes))
+    else
+      RequestStore.store[:scripts] = names.index_with(attributes)
+    end
```

Comment:
Do you think this would be a good case to use the [`.fetch`](https://github.com/steveklabnik/request_store?tab=readme-ov-file#api) API from the gem?
```suggestion
    RequestStore.fetch(:scripts) do
      names.index_with(attributes)
    end.merge!(names.index_with(attributes))
```

Reply:
I was trying to use it, but it felt not quite right to merge into itself in the empty case?

Reply:
prob fine to leave as-is